src/dpp/events/guild_member_remove.cpp

```diff
@@ -42,28 +40,19 @@ void guild_member_remove::handle(discord_client* client, json &j, const std::str
 	json d = j["d"];
 
 	dpp::guild_member_remove_t gmr(client, raw);
+	gmr.removed.fill_from_json(&(d["user"]));
+	gmr.guild_id = snowflake_not_null(&d, "guild_id");
+	gmr.removing_guild = dpp::find_guild(gmr.guild_id);
 
-	gmr.removing_guild = dpp::find_guild(snowflake_not_null(&d, "guild_id"));
-
-	if (client->creator->cache_policy.user_policy == dpp::cp_none) {
-		dpp::user u;
-		u.fill_from_json(&(d["user"]));
-		gmr.removed = &u;
-		if (!client->creator->on_guild_member_remove.empty()) {
-			client->creator->on_guild_member_remove.call(gmr);
-		}
-	} else {
-
-		gmr.removed = dpp::find_user(snowflake_not_null(&(d["user"]), "id"));
-
-		if (!client->creator->on_guild_member_remove.empty()) {
-			client->creator->on_guild_member_remove.call(gmr);
-		}
+	if (!client->creator->on_guild_member_remove.empty()) {
+		client->creator->on_guild_member_remove.call(gmr);
+	}
 
-		if (gmr.removing_guild && gmr.removed) {
-			auto i = gmr.removing_guild->members.find(gmr.removed->id);
+	if (client->creator->cache_policy.user_policy != dpp::cp_none) {
```

Comment:
This creates a lot of nesting, Couldn't we just do a big check at once?

`if (client->creator->cache_policy.user_policy != dpp::cp_none && gmr.removing_guild) {`